### PR TITLE
MCP-1816: externalize username/password

### PR DESCRIPTION
**Pull request by dimitri-amida**
# Do not hard code passwords

### What was the problem?

There are several username and passwords directly into docker and spring configuration files. These should be replaced with placeholders that, eventually, will be injected via environment variables

### How does this fix it?

- Externalized usernames and passwords into environment files and replaced them with placeholders
- Deleted configurations that are commented out or not used anywhere

### Jira Tickets

- [MCP-1816](https://amida.atlassian.net/browse/MCP-1816)

## How to test this PR

- Please regression test all endpoints to make sure nothing is broken

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

File: app/src/docker/docker-compose.yml [hunk 1]
```diff
@@ -8,22 +8,6 @@ volumes:
   pgdata:
 
 services:
-  # opa:
```
Comment (dimitri-amida) [inline app/src/docker/docker-compose.yml:11]: not used

File: app/src/docker/docker-compose.yml [hunk 2]
```diff
@@ -161,48 +134,3 @@ services:
         condition: service_healthy
     networks:
       - internal
-
-#   jaeger:
```
Comment (dimitri-amida) [inline app/src/docker/docker-compose.yml:165]: not used

File: app/src/main/resources/application.yml [hunk 3]
```diff
@@ -37,22 +37,6 @@ management:
     web.server.request.autotime.enabled: true
   server.port: 8081
 
-# Setting up K8 health probes - with health probes set to true you can use
```
Comment (dimitri-amida) [inline app/src/main/resources/application.yml:40]: not used

File: db-init/src/main/resources/flyway.conf [hunk 4]
```diff
@@ -15,15 +15,8 @@ flyway.sqlMigrationSeparator=__
 flyway.sqlMigrationSuffixes=.sql
 flyway.encoding=ISO-8859-1
 flyway.placeholderReplacement=true
-#flyway.placeholders.flywayname=example_flyway
```
Comment (dimitri-amida) [inline db-init/src/main/resources/flyway.conf:18]: not used

File: app/src/docker/docker-compose.yml [hunk 5]
```diff
@@ -150,8 +123,8 @@ services:
     image: va/abd_vro-service-data-access:latest
     environment:
       RABBITMQ_PLACEHOLDERS_HOST: rabbitmq1
-      RABBITMQ_PLACEHOLDERS_USERNAME: guest
-      RABBITMQ_PLACEHOLDERS_USERPASSWORD: guest
+      RABBITMQ_PLACEHOLDERS_USERNAME: ${RABBITMQ_USERNAME}
+      RABBITMQ_PLACEHOLDERS_USERPASSWORD: ${RABBITMQ_PASSWORD}
```
Comment (wlin2019) [inline app/src/docker/docker-compose.yml:127]: 1. Have you considered how to have the local build working after making this change? There could be a lot of this kind of "environment variables".
2. I'll update LH* as well. But 1 needs to be addressed.

**Review by wlin2019 — APPROVED** [review]
LGTM

File: app/src/docker/docker-compose.yml (hunk 5 above)
Comment (dimitri-amida) [reply app/src/docker/docker-compose.yml:127]: The local build worked for me. It's because the variables are defined in the .env file

**Comment by au2** [comment]
@wlin2019 can you take over this one as well. We need to merge develop here. When I tried locally the abd-vro and data-access containers errored out.

**Comment by wlin2019** [comment]
This PR probably should be closed. Some of its work (code changes) has been merged into the develop branch, and some of its work has been addressed in different ways in the develop branch.

**Comment by au2** [comment]
These should be all cleaned up now. We are using default rabbitmq and defaultrabbitmq passwords for now instead of creating a rabbitmq user. To be discussed later.

**Pull request closed without merging**